apps/dapp/src/components/Pages/Core/VaultPages/Stake.tsx

```diff
@@ -48,10 +46,14 @@ export const Stake = () => {
 
   const [_, refreshBalance] = useVaultBalance(vault.id);
   const [{ isLoading: refreshIsLoading }, refreshWalletState] = useRefreshWalletState();
-  const [deposit, { isLoading: depositLoading, error: depositError }] = useDepositToVault(vault.id, async () => {
-    refreshBalance();
-    refreshWalletState();
-  });
+  const [deposit, { isLoading: depositLoading, error: depositError }] = useDepositToVault(
+    vault.id,
+    async (ticker, amount) => {
```

Comment:
Aren't `ticker` and `amount` available on this page? Not sure if you really needed to pass them from the hook here, but it's fine I guess

Reply:
Technically yes... but my thinking is that they could change between the button click, and the transaction finishing. E.g. user changes the amount after submitting the txn, and we end up with the wrong amount in this success callback. So IMO it's easy enough to pass it in this way so we know exactly the amount/ticker and not rely on the existing data that's subject to change based on user input.